internal/ui/static/js/cf_component.js

```diff
@@ -0,0 +1,303 @@
+                this.results[key.value] = "";
+                this.setConfigValue = this.results;
+            }
+        }
+    }
+    handleIntegrationValue(event) {
+        const capturedEvent = event;
+        clearTimeout(this.debounce);
+            if (event.target.id.includes("integration-value")) {
+                const value = this.shadowRoot.getElementById(capturedEvent.target.id)
+                const valueuid = capturedEvent.target.id.split('-')
+                const keyval = this.shadowRoot.getElementById("integration-key-" + valueuid[2])
+                if (this.results[keyval.value] === value.value) {
+                    return;
+                } else if (value.value.trim() !== "") {
+                    this.results[keyval.value] = value.value
+                    this.setConfigValue = this.results;
+                }
+            }
+    }
+    drawView(key = "", value = "") {
+        // TODO - there is a use case when deleted category / feed remains in config
+        // we have to handle that edge case by either removing or highlight to user that entry is no longer invalid
+        const wrapper = document.createElement('div');
+        wrapper.id = 'kv-' + this.id;
+        const feed = this.feedAttr;
+        const category = this.categoryAttr;
+        const integration = this.integrationAttr;
+        const keyInput = document.createElement('select');
+        keyInput.style.width = "10rem";
+        keyInput.name = 'integration-key-' + this.id;
+        keyInput.id = 'integration-key-' + this.id;
+        if (key === "default") {
+            const defaultOption = document.createElement('option');
+            defaultOption.value = "default";
+            defaultOption.text = "default";
+            defaultOption.selected = true;
+            keyInput.appendChild(defaultOption);
+        }
+        // add input of type hidden for form submission
+        if (!document.getElementById(integration + "_config")) {
+            const result = document.createElement('input');
+            result.id = integration + "_config"
+            result.name = integration + "_config"
+            result.hidden = true;      
+            result.setAttribute("value", JSON.stringify(this.results));
+            this.parentElement.appendChild(result)
+        }
+        if (feed) {
+            const feed_value = this.feedValue
+            const option_group = document.createElement('optgroup');
+            option_group.label = "Feed"
+            feed_value.forEach((f) => {
+                const option = document.createElement('option');
+                if (`f-${f.id}` == key) option.selected = true;
+                option.value = "f-" + f.id;
+                option.text = f.title;
+                option_group.appendChild(option);
+            })
+            keyInput.appendChild(option_group);
+        }
+        if (category) {
+            const category_value = this.categoryValue.filter(obj => obj.title !== "All");
```

Comment:
filter category All since we do already have a default entry which will act as catch-all